### DISCUSSION TITLE
'x-prod' attribute for production version

### DISCUSSION THREAD

**Discussion by ASoldo** [alpinejs/alpine#3164] [3. Feature ideas]
Hi, I've been using Alpinejs for some time and I am really satisfied. My request is:

1. Add **x-prod="true/false"**
2. Shorthand **x-prod**

Example:
```html
<div x-data="{name: 'Anon'}" x-prod>
    <h1 x-text="name"></h1>
</div>
```

This flag can be used to specify what part of our app we don't want any dev tool to pick up.
Hide properties as best as can be done on client side.

I think this would make some apps more secure or just to be harder to inspect it.

Thank you for reading this message.

**Comment by ekwoka** [comment]
I think this would be impossible without a total rewrite.

A lot for he Alpine stuff is tied directly to the Dom node, as there is no vdom.

So anything sniffing can see that.

This also wouldn't affect any security, as all code in the client is assumed compromised.

Someone being able to edit it and somehow get access to things they shouldn't is not in the scope of code that only runs in the front end.

**Comment by ASoldo** [comment]
I see. I'm doing server side rendering with Go templates and Alpinejs. If I prerender and sanitize my data that I send to frontend that should be good enough?

**Reply by ekwoka** [reply]
Yes, your backend should be ensuring that no sensitive information is sent to the front and doing validation on any data it receives from the front.
